diff --git a/api/v1beta1/nmstate_types.go b/api/v1beta1/nmstate_types.go
index f3f44bb..3bec4a0 100644
--- a/api/v1beta1/nmstate_types.go
+++ b/api/v1beta1/nmstate_types.go
@@ -1,6 +1,5 @@
 /*
 
-
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
@@ -20,34 +19,25 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // NMStateSpec defines the desired state of NMState
 type NMStateSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of NMState. Edit NMState_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
-}
-
-// NMStateStatus defines the observed state of NMState
-type NMStateStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// NodeSelector is an optional selector that will be added to handler DaemonSet manifest
+	// for both workers and masters (https://github.com/nmstate/kubernetes-nmstate/blob/master/deploy/handler/operator.yaml).
+	// If NodeSelector is specified, the handler will run only on nodes that have each of the indicated key-value pairs
+	// as labels applied to the node.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 }
 
 // +kubebuilder:object:root=true
-// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=nmstates,scope=Cluster
+// +kubebuilder:storageversion
 
 // NMState is the Schema for the nmstates API
 type NMState struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   NMStateSpec   `json:"spec,omitempty"`
-	Status NMStateStatus `json:"status,omitempty"`
+	Spec              NMStateSpec `json:"spec,omitempty"`
 }
 
 // +kubebuilder:object:root=true
